packages/discount-code-generator/src/cli.js

```diff
@@ -144,13 +144,20 @@ const resolveInput = (_args) => {
         .on('data', (data) => {
           const arrayDelim = _args.multiValueDelimiter
           // Add condition so module doesn't fail if there are no cartDiscounts
-          if (data.cartDiscounts)
-          // eslint-disable-next-line no-param-reassign
+          if (data.cartDiscounts) {
+            // eslint-disable-next-line no-param-reassign
             data.cartDiscounts = data.cartDiscounts.split(arrayDelim)
-          _attributes += data
+            // eslint-disable-next-line no-param-reassign
+            data.cartDiscounts = data.cartDiscounts.map(cartDiscount => ({
+              typeId: 'cart-discount',
+              id: cartDiscount,
+            }),
+            )
+          }
+          _attributes += JSON.stringify(data)
```

Comment:
The data event is emitted only once, because the csv should contain only one row, and hence, `_attributes` should contain only one object. 
If there are multiple JSON arrays, it should throw as that would indicate an invalid file